packages/client/src/encryption/GroupKey.ts

```diff
@@ -36,12 +36,48 @@ function GroupKeyObjectFromProps(data: GroupKeyProps | GroupKeyObject): GroupKey
 
 export type GroupKeyish = GroupKey | GroupKeyObject | ConstructorParameters<typeof GroupKey>
 
-// eslint-disable-next-line no-redeclare
+/**
+ * GroupKeys are AES cipher keys, which are used to encrypt/decrypt StreamMessages (when encryptionType is AES).
+ * Each group key contains 256 random bits of key data and an UUID.
+ *
+ * A group key stores the same key data in two fields: the bytes as hex-encoded string, and as a raw Uint8Array.
+ * TODO: If this data duplication doesn't give us any performance improvement we could store the key data only
+ * in one field.
+ */
+
 export class GroupKey {
     /** @internal */
     static InvalidGroupKeyError = InvalidGroupKeyError
 
-    static validate(maybeGroupKey: GroupKey): void | never {
+    /** @internal */
+    id: string
+    /** @internal */
+    hex: string
+    /** @internal */
+    data: Uint8Array
```

Comment:
Regarding original code, could we make these readonly for clarity?
```suggestion
    readonly id: string
    /** @internal */
    readonly hex: string
    /** @internal */
    readonly data: Uint8Array
```